nano/core_test/active_transactions.cpp

```diff
@@ -967,6 +967,89 @@ TEST (active_transactions, confirmation_consistency)
 }
 }
 
+/**
+ * This test case tests the creation of reverse links for incoming confirmed blocks.
+ */
+TEST (active_transactions, reverse_link)
+{
+	// Initialise node environment and enable reverse links
+	nano::system system;
+	nano::node_config node_config = nano::node_config (nano::get_available_port (), system.logging);
+	node_config.enable_reverse_links = true;
+	auto & node = *system.add_node (node_config);
+
+	// Generate a random key pair
+	nano::keypair key;
+	// Create two blocks
+	nano::state_block_builder builder;
+	auto send1 = builder.make_block ()
+				 .account (nano::dev::genesis_key.pub)
+				 .previous (nano::dev::genesis->hash ())
+				 .representative (nano::dev::genesis_key.pub)
+				 .balance (nano::dev::constants.genesis_amount - 1)
+				 .link (key.pub)
+				 .sign (nano::dev::genesis_key.prv, nano::dev::genesis_key.pub)
+				 .work (*system.work.generate (nano::dev::genesis->hash ()))
+				 .build_shared ();
+	auto open1 = builder.make_block ()
+				 .account (key.pub)
+				 .previous (0)
+				 .representative (key.pub)
+				 .balance (1)
+				 .link (send1->hash ())
+				 .sign (key.prv, key.pub)
+				 .work (*system.work.generate (key.pub))
+				 .build_shared ();
+	// Process and confirm both blocks
+	node.process_active (send1);
+	node.process_active (open1);
+	auto vote = std::make_shared<nano::vote> (nano::dev::genesis_key.pub, nano::dev::genesis_key.prv, nano::vote::timestamp_max, nano::vote::duration_max, std::vector<nano::block_hash>{ send1->hash (), open1->hash () });
+	node.vote_processor.vote (vote, std::make_shared<nano::transport::channel_loopback> (node));
+	ASSERT_TIMELY (5s, node.ledger.cache.cemented_count == 3);
```

Comment:
I am not convinced that this is a good way to check that the blocks are cemented.
The cache counters is not synchronized with the database transactions.
The counter and the database might not be aligned.
I think using `nano::nod::get_confirmation_height`, which takes a transaction object, would be better.

Reply:
> I am not convinced that this is a good way to check that the blocks are cemented. I know we use this pattern in other places but I think this is a race condition. The cache counter is not synchronized with the database transactions. The counter and the database might not be aligned. I think using `nano::node::get_confirmation_height`, which takes a transaction object, would be better.

@dsiganos I improved the reverse link test cases after your feedback: https://github.com/Exxenoz/nano-node/commit/0a047014f824b047306aa97f5d150510723e6900

I also added nano::ledger::get_confirmation_height, because I couldn't find it somewhere else and such helpers are pretty neat.